feat: improve unit tests quality and coverage

fritznastor opened the pull request:
Closes #354

`test_flow_integration.py` improvements

Here is what I did so far:

- Switched to a Realistic Public Repository (microsoft/vscode)
- Expanded Success Path Assertions
- Added Negative and Edge Case Tests
- Rate-Limiting and Status Code Tolerance

fritznastor committed:
test_flow_integration.py improvements

filipchristiansen committed:
Merge branch 'main' into improve-unit-tests

ix-56h reviewed (changes requested):
It's a good start.
We going to merge #427 soon and we have some more typing restriction (for example `max_file_size` is a int now).

Also, don't be afraid to create a dedicated repo with weird stuff to make specific testing.

Finally, we have a lot of thing to test :
- Specific commits
- Specific tag
- Weird encoding
- Specific paths
- Blob files

fritznastor commented on tests/test_flow_integration.py:
```diff
@@ -49,7 +49,7 @@ async def test_remote_repository_analysis(request: pytest.FixtureRequest) -> Non
     client = request.getfixturevalue("test_client")
     form_data = {
-        "input_text": "https://github.com/octocat/Hello-World",
+        "input_text": "https://github.com/microsoft/vscode",
```
Are any of the example repositories on the website a good switch to use? I was thinking of maybe changing it to FastAPi or Flask

fritznastor committed:
- changed input repo from vscode to flask
-lowered max file size from 243 to 200 for all unit tests
-correctly handled missing fields and invalid token unit tests

fritznastor commented:
Hey again @ix-56h, I was able to make some hot fixes to my original additions, regarding the merge of #427 , do you want me to modify max_file_size to reflect the upcoming typing restriction from str to int? Or should I wait after you and @filipchristiansen merge #427.

Furthermore. since this is my first time contributing to an open source, I'm going to ask a lot of clarifying questions to help me contribute better! 

I need some direction/guidance on how to "create a dedicated repo with weird stuff to make specific testing." Any advice on how to begin doing this? I figured I'd start on this dedicated repo after I finished polishing up the rest of the test files that are listed as priority

fritznastor committed:
added comprehensive tests for binary files, symlinks, and file handling

fritznastor commented:
I'll wait on modifying test_git_utils.py for now until the first two files are all good.

ix-56h reviewed (changes requested):
This is pretty good so far !

fritznastor committed:
-Removed redundant unit tests
-Added pattern matching unit tests

fritznastor committed:
Improve git_utils.py test cases: added edge case, security, and input validation tests

fritznastor committed:
removed def test_create_git_command_empty_base_cmd()

fritznastor commented:
The modified test files should be good to go now, I think it's ready to be merged with main.

fritznastor reviewed (commented):
Rebased to main. No change to actual files.

ix-56h committed:
Merge branch 'main' into improve-unit-tests

fritznastor committed:
Merge branch 'main' into improve-unit-tests

fritznastor commented:
Hey @filipchristiansen, do you think you can check this out real quick, I appreciate it!

filipchristiansen commented:
> Hey @filipchristiansen, do you think you can check this out real quick, I appreciate it!

I'm away on Iceland until early next week, maybe @NicolasIRAGNE can have a look at it?

fritznastor committed:
modified unit test test_repository_with_patterns to handle different repositories dynamically

fritznastor commented:
Hey @NicolasIRAGNE , thanks for getting back to me.

Specifically:
Should I be mocking all external repo access and simulating Git responses instead of using real repos like flask or vscode?

Is there an internal fixture or preferred test structure you'd like me to follow when testing edge cases like binary files, symlinks, etc.?

Lastly, would it be more helpful for me to focus on unit tests with mocked I/O rather than higher-level integration scenarios?

I’m happy to do a rewrite that better fits the project’s goals—I want to make sure I’m going about it the right way.
